### ISSUE TITLE
Missing history?

### ISSUE THREAD

**Issue by ylluminate** [pawurb/lazyme#1]
Just saw your article and gave your tool a run. Oddly it doesn't show anything very meaningful as all of the suggestions are very infrequently used commands in my ZSH history. Wondering if I'm using something that's messing up your scripts ability to pull this data out properly.

**Comment by pawurb** [comment]
This gem is a very simple ruby script, which parses  `.zsh_history` file, and groups lines by frequency. If you don't have any sensitive data there you could send me this file and I will check if anything is odd. 

Remember that script only groups identical commands together so `cd Docs` and `cd Stuff` will not be matched.

**Comment by ylluminate** [comment]
Sorry, somehow missed your message here.  I don't really want to send my history, BUT I can see that it's because my history is stored in ~/.zsh/history.zsh.  I guess the trick is that we need to use `$HISTFILE` instead of an absolute path for history file location.

**Comment by pawurb** [comment]
Looks like $HISTFILE is not available to the Ruby process:

```
`$HISTFILE` => ""
```

You have some other idea on how to make it more robust ?

**Comment by westonganger** [comment]
Can't you access environment variables with `ENV['HISTFILE']`

**Comment by sshaw** [comment]
> Looks like $HISTFILE is not available to the Ruby process:

Under Bash (not sure about zsh) `HISTFILE` will be set to the default value, so it will be set in interactive shells.  This does not `export` it to ~subshells~ child processes though. That must be done explicitly. 

For example:

```
~/code/js/prototoy >echo $HISTFILE
/Users/sshaw/.bash_history
~/code/js/prototoy >export -p  | grep HIST
declare -x HISTCONTROL="ignoreboth"
declare -x HISTFILESIZE="1000"
declare -x HISTSIZE="5000"
declare -x IRB_HISTCONTROL="ignoreboth"
declare -x IRB_HISTIGNORE="q!:^\\s*r\\b"
~ >ruby -e'p ENV["HISTFILE"]'
nil
```

But when exported:
```
~/code/js/prototoy >export HISTFILE
~/code/js/prototoy >export -p  | grep 'HISTFILE\>'
declare -x HISTFILE="/Users/sshaw/.bash_history"
~/code/js/prototoy >ruby -e'p ENV["HISTFILE"]'
"/Users/sshaw/.bash_history"
```

**Comment by pawurb** [comment]
That's correct but I don't know how I could integrate it into script execution.

I tried to change bin/lazyme into:

```
export HISTFILE
ruby lazyme.rb
```

but HISTFILE gets exported as an empty variable. You can check it out on fix_histfile branch.

**Comment by ylluminate** [comment]
This seems to work for me in `zsh` @pawurb. I'm actually getting output now after checking out that branch and executing: `export HISTFILE; ruby lazyme.rb`

**Comment by pawurb** [comment]
Does not work for me. You must have exported HISTFILE before executing the command.

I mean you should get it working by running:

```
cd bin
./lazyme
```

because we need the whole thing to work as a script not requiring user to run more complex commands.

**Comment by sshaw** [comment]
> You must have exported HISTFILE before executing the command.

Yes. This is the point, sorta. If the user has a custom history file they will `export HISTFILE`. Otherwise the shell will use its default, which you're checking for. 


Though there's an edge case: if the user runs (or has ran) bash and zsh what does the program summarize?

You may want to add a `--shell` option or accept an optional argument specifying the file.

**Comment by pawurb** [comment]
Added an option to provide a custom history file path and released as 1.1.0. Hope it solves the problem.
